### ISSUE TITLE
Make website error notifications more verbose

### ISSUE THREAD

**Issue by wendelap** [OpenEnergyDashboard/OED#380]
Error notifications in the upper right corner of the website ("Failed to create group" and "Failed to upload readings") are incredibly vague. It'd be more user-friendly to include a brief description of why it failed or at least an error code that could be looked up.

**Comment by NoraCodes** [comment]
I definitely agree with this. I think the best way to handle it would be to have a `fail()` method that can be called any time a route fails, sending the full error to the client, where it can be handled appropriately, as well as logging it.

**Comment by huss** [comment]
Many of these have been addressed. We will do a careful sweep for the next release.

**Comment by huss** [comment]
This is also discussed in #869. Given this and a lot of this has been addressed, this issue is being closed.
